pmg/models/resources.py

```diff
@@ -1333,16 +1333,28 @@ def annual_attendance_trends(cls, from_year, to_year):
             .all()
 
     @classmethod
-    def annual_attendance_trends_for_committee(cls, committee_id):
-        subquery = db.session.query(
-            func.date_part('year', CommitteeMeeting.date).label('year'),
-            func.count(case([(CommitteeMeetingAttendance.attendance.in_(CommitteeMeetingAttendance.ATTENDANCE_CODES_PRESENT), 1)])).label('n_present'),
-            func.count(CommitteeMeetingAttendance.id).label('n_members')
-        )\
-            .group_by('year', CommitteeMeeting.id)\
-            .filter(CommitteeMeeting.committee_id == committee_id)\
-            .filter(CommitteeMeetingAttendance.meeting_id == CommitteeMeeting.id)\
-            .subquery('attendance')
+    def committee_attendence_trends(cls, committee_id, period):
+        if period == 'current':
+            subquery = db.session.query(
+                func.date_part('year', CommitteeMeeting.date).label('year'),
+                func.count(case([(CommitteeMeetingAttendance.attendance.in_(CommitteeMeetingAttendance.ATTENDANCE_CODES_PRESENT), 1)])).label('n_present'),
```

Comment:
thsi is all the same except for the date, right? Can't we assign the date to a variable and then have this code once?

Reply:
Not really, query is filtered differently based on whether is historical or not.